packages/flutter/lib/fix_data.yaml

```diff
@@ -17,6 +17,100 @@
 
 version: 1
 transforms:
+  # Changes made in https://github.com/flutter/flutter/pull/108052
+  - title: 'Remove BoxShape.rectangle'
+    date: 2022-08-03
+    element:
+      uris: [ 'painting.dart', 'rendering.dart', 'widgets.dart', 'material.dart', 'cupertino.dart' ]
+      method: 'copyWith'
+      inClass: 'BoxDecoration'
+    oneOf:
+      - if: "shape == 'BoxShape.rectangle'"
+        changes:
+          - kind: 'removeParameter'
+            name: 'shape'
+    variables:
+      shape:
+        kind: 'fragment'
+        value: 'arguments[shape]'
+  # Changes made in https://github.com/flutter/flutter/pull/108052
+  - title: 'Remove BoxShape.rectangle'
+    date: 2022-07-27
+    element:
+      uris: [ 'painting.dart', 'rendering.dart', 'widgets.dart', 'material.dart', 'cupertino.dart' ]
+      inClass: 'BoxDecoration'
+      constructor: ''
+    oneOf:
+    - if: "shape == 'BoxShape.rectangle'"
+      changes:
+        - kind: 'removeParameter'
+          name: 'shape'
+    variables:
+      shape:
+        kind: 'fragment'
+        value: 'arguments[shape]'
+  # Changes made in https://github.com/flutter/flutter/pull/108052
+  - title: 'Replace with ShapeDecoration'
+    date: 2022-07-27
+    element:
+      uris: [ 'painting.dart', 'rendering.dart', 'widgets.dart', 'material.dart', 'cupertino.dart' ]
+      inClass: 'BoxDecoration'
+      constructor: ''
+    oneOf:
+    - if: "shape == 'BoxShape.circle' && border == '' && backgroundBlendMode == ''"
+      changes:
+        - kind: 'replacedBy'
+          newElement:
+            uris: [ 'painting.dart', 'rendering.dart', 'widgets.dart', 'material.dart', 'cupertino.dart' ]
+            inClass: 'ShapeDecoration'
+            constructor: ''
+        - kind: 'addParameter'
+          index: 1
+          name: 'shape'
+          style: optional_named
+          argumentValue:
+            expression: '{% CircleBorder %}()'
+            requiredIf: "shape == 'BoxShape.circle'"
+            variables:
+              CircleBorder:
+                kind: 'import'
+                uris: [ 'painting.dart', 'rendering.dart', 'widgets.dart', 'material.dart', 'cupertino.dart' ]
+                name: 'CircleBorder'
+        - kind: 'removeParameter'
+          name: 'boxShadow'
```

Comment:
`boxShadow` is named `shadows` in `ShapeDecoration` api so we need to migrate that too when we convert `BoxDecoration` to `ShapeDecoration`.  If `boxShadow` is not in use then it won't be added as `shadow`. 

> This fix rule is only applied if shape == 'BoxShape.circle' && border == '' && backgroundBlendMode == ''.. is box shadow something that should not be here?

This is because 1. we only apply this fix if the shape is `BoxShape.circle` 2. ShapeDecoration doesn't have `border` and `backgroundBlendMode` params so we only apply if they don't exist in the code.

On second thought, `CircleBorder` can have borderSide but the conversion from `border` would be hard and complicated so I don't know what else to do here.

Reply:
IMO you can ignore backgroundBlendMode, it's probably rarely being used and I don't think will break any test.

The only allowed Border with circle is Border.all(), so that could become BorderSide with the same parameters.